<launch>
  <include file="$(find autorally_core)/launch/hardware.machine" />

  <node pkg="nodelet" type="nodelet" name="ServoInterface" args="load autorally_core/ServoInterface autorally_core_manager" machine="autorally-master" output="screen">
<!--  <node name="servoInterface" pkg="autorally_core" type="servoInterface" machine="autorally-master" output="screen"> -->
    <param name="controllerType" value="micro"/>
    <param name="servoCommandRate" value="75" />
    <param name="throttleBrakeCoupled" value="true" />
    <param name="servoCommandMaxAge" value="0.2" />

    <rosparam param="servos" command="load" file="$(env AR_CONFIG_PATH)/arChassisConfig_$(env AR_CHASSIS).yaml" />
    <!-- <rosparam param="throttleCalibration" command="load" file="$(env AR_CONFIG_PATH)/throttlePositionCalibration.yaml" /> -->

    <rosparam param="servoCommandProirities" command="load" file="$(env AR_CONFIG_PATH)/chassisCommandPriorities.yaml" />
    <!--configure settings for 57600 baud, 8N1 -->
    <param name="port" value="/dev/arServoController" />
    <param name="serialBaud" value="57600" />
    <param name="serialDataBits" value="8" />
    <param name="serialParity" value="none" />
    <param name="serialStopBits" value="1" />
    <param name="serialHardwareFlow" value="false" />
    <param name="serialSoftwareFlow" value="false" />
  </node>

</launch>
